fotoapparat/src/main/java/io/fotoapparat/routine/focus/ManualFocusTask.java

```diff
@@ -0,0 +1,39 @@
+package io.fotoapparat.routine.focus;
+
+import android.graphics.Rect;
+import android.support.annotation.NonNull;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.FutureTask;
+
+import io.fotoapparat.hardware.CameraDevice;
+import io.fotoapparat.result.FocusResult;
+
+public class ManualFocusTask extends FutureTask<FocusResult> {
+
+    public ManualFocusTask(final CameraDevice cameraDevice,
+                           final Rect cameraViewRect,
+                           final float focusX, final float focusY) {
+        super(new Callable<FocusResult>() {
+            @Override
+            public FocusResult call() throws Exception {
+                return performManualFocus(cameraDevice, cameraViewRect, focusX, focusY);
+            }
+        });
+    }
+
+    @NonNull
+    private static FocusResult performManualFocus(CameraDevice cameraDevice,
+                                                  Rect cameraViewRect,
+                                                  float focusX, float focusY) {
+        cameraDevice.setFocusArea(cameraViewRect, focusX, focusY);
+        boolean isSucceeded = cameraDevice.autoFocus().succeeded;
+        if (isSucceeded) {
+            cameraDevice.cancelAutoFocus();
```

Comment:
Do we need to cancel it?

Reply:
Yeah, because it's still autofocusing without this.

Reply:
Yeah, because it's still autofocusing without this.